src/components/SingleArticle.jsx

```diff
@@ -1,28 +1,46 @@
 import { useEffect, useState } from "react";
-import { fetchArticleById } from "../utils/utils";
+import { fetchArticleById, fetchCommentsByArticleId } from "../utils/utils";
 import { useParams } from "react-router-dom";
+import CommentCard from "./CommentCard";
 import "../App.css";
 
 export default function SingleArticle() {
     const [singleArticle,  setSingleArticle] = useState({})
+    const [commentsByArticleId, setCommentsByArticleId] = useState([])
     const [isLoading, setIsLoading] = useState(true)
     const {article_id}=useParams()
     
     useEffect(()=>{
         fetchArticleById(article_id).then((article)=>{
             setSingleArticle(article)
             setIsLoading(false)
+            return article
+        }).then((article)=> {
+            fetchCommentsByArticleId(article.article_id).then((comments)=>{
+                setCommentsByArticleId(comments)
+            })
```

Comment:
This wouldn't necessarily need to be in a then chain as you have access to the article id, and this has lead to nested then blocks.